### ISSUE TITLE
Errors on import while running jest

### ISSUE THREAD

**Issue by noire-munich** [redwoodjs/redwood#2890]
After upgrading to v0.34.0 I've found myself with all my tests failing on this kind of error:

```cli
 FAIL   api  api/src/services/products/products.test.ts
  ● Test suite failed to run

    Cannot find module '../../../../src/lib/db' from 'src/services/products/products.test.ts'

> 1 | import { db } from 'src/lib/db'
        | ^
      2 | import { disciplines } from 'src/services/disciplines/disciplines'
      3 | import { levels } from 'src/services/levels/levels'
      4 | import {

      at Resolver.resolveModule (../node_modules/jest-resolve/build/index.js:306:11)
      at Object.<anonymous> (src/services/products/products.test.ts:1:1)

```

The error in itself is that imports which use to work no longer work, eventhough files have not changed. This happens within tests files but also with imports that are in regular services. Thoses services work well with dev server started.

I've ran tests on a new project and it doesn't seem to be related to v0.34.0 on new installs, but it could happen on upgrading projects.

Here are some relevant files:
```js
// api/jest.config.js
module.exports = require('@redwoodjs/testing/config/jest/api')
```

```js
// api/jsconfig.json
{
  "compilerOptions": {
    "noEmit": true,
    "allowJs": true,
    "esModuleInterop": true,
    "target": "esnext",
    "module": "esnext",
    "moduleResolution": "node",
    "baseUrl": "./",
    "rootDirs": [
      "./src",
      "../.redwood/types/mirror/api/src"
    ],
    "paths": {
      "src/*": [
        "./src/*",
        "../.redwood/types/mirror/api/src/*"
      ],
      "types/*": [
        "./types/*"
      ]
    },
    "typeRoots": [
      "../node_modules/@types",
      "./node_modules/@types"
    ],
    "types": [
      "jest"
    ]
  },
  "include": [
    "src",
    "../.redwood/types/includes/all-*",
    "../.redwood/types/includes/api-*"
  ]
}
```

```js
// api/tsconfig.json
{
  "compilerOptions": {
    "noEmit": true,
    "allowJs": true,
    "esModuleInterop": true,
    "target": "esnext",
    "module": "esnext",
    "moduleResolution": "node",
    "baseUrl": "./",
    "rootDirs": [
      "./src",
      "../.redwood/types/mirror/api/src"
    ],
    "paths": {
      "src/*": [
        "./src/*",
        "../.redwood/types/mirror/api/src/*"
      ],
      "types/*": [
        "./types/*"
      ]
    },
    "typeRoots": [
      "../node_modules/@types",
      "./node_modules/@types"
    ],
    "types": [
      "jest"
    ]
  },
  "include": [
    "src",
    "../.redwood/types/includes/all-*",
    "../.redwood/types/includes/api-*",
    "types/*.d.ts"
  ]
}
```

```js
// babel.config.js
module.exports = {
  presets: ['@redwoodjs/core/config/babel-preset'],
}
```

And finally, the test file:
```js
import { db } from 'src/lib/db'
import { disciplines } from 'src/services/disciplines/disciplines'
import { levels } from 'src/services/levels/levels'
import {
  attributeDefaultMonitorToCourse,
  courseDefaultMonitor,
} from 'src/services/monitors/monitors'
import { productStub } from 'src/services/products/utilities'
import {
  createCourse,
  createProduct,
  product,
  products,
  updateProduct,
} from './products'

describe('products', () => {
  scenario('Retrieve a product', async (scenario) => {
    mockCurrentUser(scenario.user.manager)

    const p = await product({ id: scenario.product.one.id })

    expect(p.label).toMatch('Product label')

    expect(p.basePrice).toEqual(30)
  })

/** I'll spare you the other tests. */

})

```

**Comment by thedavidprice** [comment]
Hi @noire-munich Let's see if we can get you sorted.

Possibly same issue, re: Code Mods, as this https://github.com/redwoodjs/redwood/issues/2871#issuecomment-866105325?

**Comment by peterp** [comment]
If this is the case then I introduced this by accident. I meant for it to continue working, but to display a deprecation warning

**Comment by peterp** [comment]
@thedavidprice if this is the case, please assign to me, and I'll make sure that it works as expected

**Comment by noire-munich** [comment]
Hi @thedavidprice 
Thanks, it doesn't seem to be the same issue. I removed node_modules, installed again, same bug pops up.
Once the test process is done ( can't force quit apparently when it's buggy ), i'll try with `install --force` for the sake of it & report.

```
$ /Workspace/Code/SportOffice/node_modules/.bin/rw info

  System:
    OS: Linux 5.11 Ubuntu 21.04 (Hirsute Hippo)
    Shell: 5.8 - /usr/bin/zsh
  Binaries:
    Node: 14.0.0 - /tmp/yarn--1624640146458-0.23422278367246463/node
    Yarn: 1.22.5 - /tmp/yarn--1624640146458-0.23422278367246463/yarn
  Browsers:
    Chrome: 91.0.4472.114
    Firefox: 89.0.1
  npmPackages:
    @redwoodjs/core: ^0.34.1 => 0.34.1 

Done in 11.37s.
```

**Comment by noire-munich** [comment]
Ok, I confirm with little surprise, `install --force` does not resolve the issue.

**Comment by thedavidprice** [comment]
@noire-munich thank you for reporting this! I've labeled as a bug and queued up for priority.

@peterp done! Thanks.

**Comment by cjreimer** [comment]
FYI, I am finding this same behavior as well relating to tests.  I'm pretty sure I saw it in the redwood 0.33 series as well, and maybe even before then.  
Note:
- The imports work fine when building or running using `yarn rw dev`.  
- I have completed the changes to jest.config.js as required by redwood 0.34.
- I am on Windows, so there is a chance this is a Windows thing (behaviour consistent on both powershell and bash shells).

An example from my web side tests is below.  

```
 FAIL  web/src/components/Comments/Comments.test.js
  ● Test suite failed to run

    Cannot find module '../../src/layouts/MainAppLayout/MainAppLayout' from 'src/Routes.js'

    Require stack:
      src/Routes.js
      D:/CENGYS/Dev/cengys-r/node_modules/@redwoodjs/testing/dist/MockProviders.js
      D:/CENGYS/Dev/cengys-r/node_modules/@redwoodjs/testing/dist/customRender.js
      D:/CENGYS/Dev/cengys-r/node_modules/@redwoodjs/testing/dist/index.js
      D:/CENGYS/Dev/cengys-r/node_modules/@redwoodjs/testing/config/jest/web/jest.setup.js

      10 | import { Router, Route, Private, Set } from '@redwoodjs/router'
    > 12 | import MainAppLayout from 'src/layouts/MainAppLayout/MainAppLayout'
         | ^
      13 | import MainFrontLayout from 'src/layouts/MainFrontLayout/MainFrontLayout'
      14 |
      15 | const userRouteParamTypes = {

      at Resolver.resolveModule (../node_modules/jest-resolve/build/index.js:306:11)
      at Object.<anonymous> (src/Routes.js:12:1)


Test Suites: 68 failed, 68 total
Tests:       0 total
Snapshots:   0 total
Time:        50.399 s
Ran all test suites.

```

**Comment by nnennajohn** [comment]
Hi. I can confirm this problem still persists. I am on redwood 0.34.1. Is there a temp workaround to resolving the path for jest?

Thank you.

<img width="1393" alt="Screen Shot 2021-07-11 at 1 20 03 AM" src="https://user-images.githubusercontent.com/1761197/125187807-4f0e5880-e1e6-11eb-8946-4b5aa52c8d42.png">

**Comment by nnennajohn** [comment]
This seems to be a problem with the moduleNameMapper. Worked around it by overwriting it. The below is in the api folder. Probably the same in the web folder, as all my tests were failing after update.

```
const redwoodjsJestConfig = require('@redwoodjs/testing/config/jest/api')

module.exports = {
  ...redwoodjsJestConfig,
  moduleNameMapper: {
    ...(redwoodjsJestConfig.moduleNameMapper || {}),
    'src/(.*)': '<rootDir>/src/$1',
  },
}
```
<img width="707" alt="Screen Shot 2021-07-11 at 1 28 33 AM" src="https://user-images.githubusercontent.com/1761197/125188183-b37de780-e1e7-11eb-9ec4-886fdc042ac6.png">

**Comment by noire-munich** [comment]
Thanks @nnennajohn , that one works for me on v0.35.0.

**Comment by peterp** [comment]
I've got a fix for this in #2938

**Comment by cjreimer** [comment]
Thanks to @nnennajohn, the above workaround improves things for me..  However, jest still does not address directory named imports.

For example:

`import Comments from 'src/components/Comments/Comments'`
works but
`import Comments from 'src/components/Comments'`

in the various source files causes issues.  I see there's an old package out there that seems to be targetted at this issue:
[https://www.npmjs.com/package/jest-directory-named-resolver](https://www.npmjs.com/package/jest-directory-named-resolver)

Is it just me, or do others have the same issue?

**Comment by nnennajohn** [comment]
@cjreimer I ran into the same problem, so you are not alone. :) 

If you used the hack above, then the problem arises, because in the line of code below,
```'src/(.*)': '<rootDir>/src/$1',```
src/cool will try to find <rootDir>/src/cool.ts, not <rootDir>/src/cool/cool.ts
I did not have time to further investigate this, so ended up just creating an index file at root of each component to re-export the default export in the Component file, as I actually am not too fond of the Comment/Comment import myself. This way, it will work with either style of import.

Either way, the hack above should not be the solution. Will keep an eye out for @peterp 's PR and try again once it's fixed.
